src/python/pants/build_graph/target.py

```diff
@@ -794,12 +760,44 @@ def strict_dependencies(self, dep_context):
     """
     strict_deps = self._cached_strict_dependencies_map.get(dep_context, None)
     if strict_deps is None:
+      closure_kwargs = dep_context.target_closure_kwargs
+      respect_intransitive = closure_kwargs.get('respect_intransitive')
+      include_scopes = closure_kwargs.get('include_scopes')
+      exclude_scopes = closure_kwargs.get('exclude_scopes')
+      def pred(target, dep):
```

Comment:
Worth extracting and calling via `functools.partial(self._strict_deps_predicate, dep_context)` ?

Reply:
I could try that. I do like having the predicate's definition close to use though.